src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadManifest.targets.in

```diff
@@ -41,8 +41,18 @@
   <PropertyGroup Condition="'$(RuntimeIdentifier)' == 'browser-wasm' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
     <!-- $(WasmBuildNative)==true is needed to enable workloads, when using native references, without AOT -->
     <!-- FIXME: is the blazor condition here correct? -->
-    <_WasmNativeWorkloadNeeded Condition="'$(RunAOTCompilation)' == 'true' or '$(WasmEnableSIMD)' == 'true' or '$(WasmEnableLegacyJsInterop)' == 'false' or '$(WasmBuildNative)' == 'true' or
-      '$(WasmGenerateAppBundle)' == 'true' or '$(_UsingBlazorOrWasmSdk)' != 'true'" >true</_WasmNativeWorkloadNeeded>
+    <!-- Keep in sync with WasmApp.Native.targets -->
+    <_WasmNativeWorkloadNeeded Condition="
```

Comment:
Does this get evaluated late enough that defaults have already been applied for all these properties? If not, shouldn't most of these be of the form `!= 'false'` instead of `== 'true'`?

Reply:
Good question! Setting `_WasmNativeWorkloadNeeded` will cause the build to fail if the user does not have the `wasm-tools` workload installed.

- The workload is needed for doing any native build, which is when:
   - If the various settings *don't* match, what the bits in the runtime pack were built with (for example `WasmEnableLegacyJsInterop=false`)
   - Using native references, or AOT
   - Explicitly asked for, by setting `WasmBuildNative=true`

- But without the workload the targets for these native builds are not available at all, so essentially, the user will be expecting a native build but nothing would happen.
- To prompt the user to install the the workload we set `_WasmNativeWorkloadNeeded=true` here, and essentially try to determine whether a native build might be needed.
- We don't want it to get triggered by default
- So, most of these properties are for settings different from the runtime pack build.

But, having said that, I'm now curious about `WasmEnableSIMD`, and `WasmEnableExceptionHandling`, which are `true` for the runtime pack, and thus AFAIU, the conditions should be inverted. I will check this more, and get back to you.

Reply:
And to answer the original question, we want to explicitly checking for a value that is not the default. And that's why this list needs to be in sync with what we have `WasmApp.Native.targets, what `wasm.proj` uses for building the runtime pack bits.

Reply:
So, the `WasmEnableSIMD`, and `WasmEnableExceptionHandling` were incorrect. Apparently, we hadn't changed them since they were made the default.